variable/include/scipp/variable/bucket_variable.tcc

```diff
@@ -115,12 +115,22 @@ public:
   units::Unit elem_unit(const VariableConstView &var) const override {
     return std::get<2>(var.constituents<bucket<T>>()).unit();
   }
+  void expect_can_set_elem_unit(const VariableView &var,
+                                const units::Unit &u) const override {
+    if constexpr (std::is_same_v<T, VariableConstView>)
+      throw std::runtime_error("Cannot set unit via const non-owning view");
```

Comment:
Does this error message make sense? It seems like if you pass it `T = VariableView`, there would be no exception. I realise that this is not the only place with this message but would it make sense to change it to refer to the constness?

Reply:
Yes, this is about `VariableConstView`, it is ok to change for `VariableView`. Non-owning views are currently only used as implementation details for some operation, so it is unlikely that users ever see the message.